### ISSUE TITLE
Allow disabling cert genration for the ingress

### ISSUE THREAD

**Issue by rtyler** [goharbor/harbor-helm#229]
I am looking to use harbor in an environment which uses cert-manager and Let's Encrypt for certificates.

It's not clear to me how to disable properties like `expose.tls.secretName` to avoid the generation of a self-signed certificate, in order to allow the cluster issuer for cert-manager to do its job.

**Comment by rtyler** [comment]
The workaround that I used was to provision with TLS disabled, and then manually patch the ingress resource to add the `tls` entries for my cert-issuer to work.

**Comment by PaulFarver** [comment]
Not a maintainer, but just throwing my two cents in here.
The ingress secret should only be generated, if the `expose.tls.secretName` evaluates to false, for example if it is not set.

from templates/_helpers.tpl
```
...
{{- define "harbor.autoGenCert" -}}
  {{- if and .Values.expose.tls.enabled (not .Values.expose.tls.secretName) -}}
    {{- printf "true" -}}
  {{- else -}}
    {{- printf "false" -}}
  {{- end -}}
{{- end -}}

{{- define "harbor.autoGenCertForIngress" -}}
  {{- if and (eq (include "harbor.autoGenCert" .) "true") (eq .Values.expose.type "ingress") -}}
    {{- printf "true" -}}
  {{- else -}}
    {{- printf "false" -}}
  {{- end -}}
{{- end -}}
...
```

from templates/ingress/secret.yaml
```
{{- if eq (include "harbor.autoGenCertForIngress" .) "true" }}
...
{{- end }}
```

When you want cert-manager to generate a secret for you, you create an ingress with the annotation `kubernetes.io/tls-acme: "true"` and set the secretName in the tls section to something. cert-manager should discover that the secret doesn't have a certificate (because it doesn't exist), and automatically create a new one, and attempt to get a certificate from Let's Encrypt.

So all you need to do is set `expose.tls.secretName` to something like `my-harbor-tls` and add `kubernetes.io/tls-acme: "true"` to `expose.ingress.annotations`

It could be more obvious in the documentation, though.

**Comment by phenixblue** [comment]
> The workaround that I used was to provision with TLS disabled, and then manually patch the ingress resource to add the `tls` entries for my cert-issuer to work.

Same here. I use a wildcard cert as the default TLS backend on my Ingress Controller. Would be great to have support for these scenarios in the Helm chart.

**Comment by reasonerjt** [comment]
I assume this has been fixed as we allow referencing an existing secret for the certificate?

**Comment by ywk253100** [comment]
Fixed by #671 which introduces the option `expose.tls.certSource` to set the source of certificate
